pkg/controller/federatedcluster/controller_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federatedcluster
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/common"
+	fedv1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
+	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+)
+
+func TestThresholdCheckedClusterStatus(t *testing.T) {
+
+	epoch := metav1.Now()
+	t1 := metav1.Time{Time: epoch.Add(1 * time.Second)}
+	t2 := metav1.Time{Time: epoch.Add(2 * time.Second)}
+	t3 := metav1.Time{Time: epoch.Add(3 * time.Second)}
+	t4 := metav1.Time{Time: epoch.Add(4 * time.Second)}
+	t5 := metav1.Time{Time: epoch.Add(5 * time.Second)}
+
+	config := util.ClusterHealthCheckConfig{
+		PeriodSeconds:    10 * time.Second,
+		FailureThreshold: 3,
+		SuccessThreshold: 1,
+		TimeoutSeconds:   3 * time.Second,
+	}
+
+	testCases := map[string]struct {
+		clusterStatus         *fedv1a1.FederatedClusterStatus
+		store                 *ClusterStore
+		expectedClusterStatus *fedv1a1.FederatedClusterStatus
+	}{
+		"ClusterReadyAtBegining": {
+			clusterStatus:         clusterStatus(common.ClusterReady, corev1.ConditionTrue, t1, t1),
```

Comment:
nit: Specifying `common.ClusterReady` seems entirely redundant given that every call to `clusterStatus` provides the same value.

Reply:
Done